### PR TITLE
Add momentum optimizer option to speech_commands

### PR DESCRIPTION
**Pull request by mswang12**
- Add command line argument to choose momentum optimizer option to speech_commands.
- Default optimizer for speech commands is unchanged to still use GradientDescentOptimizer.
- Default values for momentum optimizer are to use a momentum of .9 and use_nesterov=True.

### CODE REVIEW

**Comment by MarkDaoust** [comment]
This looks reasonable. Any comment on when/why someone should choose this option? Does it work better for some configurations?

**Comment by mswang12** [comment]
@MarkDaoust : For the audio dataset we're working with, the Nesterov Momentum Optimizer gets a better training/validation accuracy after the same number of steps compared with the standard GradientDescentOptimizer. Would recommend users to try this option if they're trying to improve audio recognition on their custom datasets!

**Comment by mswang12** [comment]
@MarkDaoust : Can you approve this pull request or are there changes you'd like me to make to the code? Thanks!

**Review by MarkDaoust — DISMISSED** [review]
Sorry about the delay. 

I think we're good here, I just fixed it to use `tf.compat.v1` since the TF2.0 package is coming out soon, and  this will be broken otherwise.

**Comment by mswang12** [comment]
Awesome thanks!

**Comment by mswang12** [comment]
@MarkDaoust : Fixed a line that was over 80 characters to be less than 80 characters long. Can you approve this pull request again? Thanks!

**Review by MarkDaoust — DISMISSED** [review]

**Comment by MarkDaoust** [comment]
Hi,

It looks like this is failing the tests.

https://source.cloud.google.com/results/invocations/1dc48548-20e7-4cb6-9759-e406882e2228/targets/%2F%2Ftensorflow%2Fexamples%2Fspeech_commands:train_test/tests

I think you just need to also define this flag in `trail_test.py` which also runs this code.

**Comment by mswang12** [comment]
Sorry didn't realize the tests needed to be modified and will fix as soon as I can. Thanks for letting me know!

**Comment by mswang12** [comment]
@MarkDaoust : Sorry for the delay. I've added the optimizer flag to train_test.py. Can you take another look at this PR? Thanks!

Verified that the following tests now pass:
> bazel test --jobs=2 --ram_utilization_factor=30 //tensorflow/examples/speech_commands:train_test

**Comment by mswang12** [comment]
Tried resolving a merge commit through the web browser but seems like Googlebot is no longer happy with the CLA we signed.

**Comment by rthadur** [comment]
Seems auto-merge is not happening but the changes are now committed so we can close this. Thank you for the PR.

**Comment by mswang12** [comment]
Thanks!
